contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+    /// @param _lowerBoundingTick Smaller bounding tick of our liquidity position
+    /// @param _upperBoundingTick Larger bounding tick of our liquidity position
+    /// @param _tickClosestToParity Tick that is closer to 1:1 price parity
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        address _wethAddress,
+        address _OETHbAddress,
+        address _swapRouter,
+        address _nonfungiblePositionManager,
+        address _clPool,
+        address _sugarHelper,
+        int24 _lowerBoundingTick,
+        int24 _upperBoundingTick,
+        int24 _tickClosestToParity
+    ) InitializableAbstractStrategy(_stratConfig)
+    {
+        WETH = _wethAddress;
+        OETHb = _OETHbAddress;
+        swapRouter = ISwapRouter(_swapRouter);
+        positionManager = INonfungiblePositionManager(
+            _nonfungiblePositionManager
+        );
+        clPool = ICLPool(_clPool);
+        helper = ISugarHelper(_sugarHelper);
+        sqrtRatioX96Tick0 = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(_lowerBoundingTick);
+        sqrtRatioX96Tick1 = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(_upperBoundingTick);
+        sqrtRatioX96TickClosestToParity = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(_upperBoundingTick);
+
+        lowerTick = _lowerBoundingTick;
+        upperTick = _upperBoundingTick;
+        tickSpacing = 1;
+    }
+
+    /**
+     * @notice initialize function, to set up initial internal state
+     * @param _rewardTokenAddresses Address of reward token for platform
+     * @param _assets Addresses of initial supported assets
+     * @param _pTokens Platform Token corresponding addresses
+     * @param _clGauge Address of the Aerodrome slipstream pool gauge
+     */
+    function initialize(
+        address[] memory _rewardTokenAddresses,
+        address[] memory _assets,
+        address[] memory _pTokens,
+        address _clGauge
+    ) external onlyGovernor initializer {
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+        // TODO: move to constructor once this is known on the mainnet
+        clGauge = ICLGauge(_clGauge);
+    }
+
+    /***************************************
+                  Configuration 
+    ****************************************/
+
+    /**
+     * TODO: delete once we get the gauge.
+     */
+    function setGauge(address _clGauge) external onlyGovernor {
+        clGauge = ICLGauge(_clGauge);
+    }
+
+    /**
+     * @notice Set the new desired WETH share
+     * @param _amount The new amount specified in basis points
+     */
+    function setPoolWethShare(uint256 _amount) external onlyGovernor {
+        // TODO tests:
+        // - governor can update
+        // - non governor can not update
+        // - must be within allowed values (event emitted)
+
+        require(_amount < 1e18, "Invalid poolWethShare amount");
```

Comment:
The other parts of the code revert if `poolWethShare` is zero. Would be nice to add that check here

Reply:
agreed: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/f269553d634fe1639e8d2c3bd3a08d3aa5a0650d